reasoner/resolution/framework/Request.java

```diff
@@ -95,4 +98,86 @@ public String toString() {
                 '}';
     }
 
+    public boolean isToSubsumed() {
+        return false;
+    }
+
+    public ToSubsumed asToSubsumed() {
+        throw TypeDBException.of(ILLEGAL_STATE);
+    }
+
+    public boolean isToSubsumer() {
+        return false;
+    }
+
+    public ToSubsumer asToSubsumer() {
+        throw TypeDBException.of(ILLEGAL_STATE);
+    }
+
+    public static class ToSubsumed extends Request {
```

Comment:
These are the extra request types that can tell a resolver where to go next for subsumption answers